cdap-app-fabric/src/main/java/io/cdap/cdap/scheduler/ProgramScheduleService.java

```diff
@@ -60,13 +62,48 @@
   private final AuthorizationEnforcer authorizationEnforcer;
   private final AuthenticationContext authenticationContext;
   private final Scheduler scheduler;
+  private final TimeSchedulerService timeSchedulerService;
 
   @Inject
   ProgramScheduleService(AuthorizationEnforcer authorizationEnforcer,
-                         AuthenticationContext authenticationContext, Scheduler scheduler) {
+                         AuthenticationContext authenticationContext, Scheduler scheduler,
+                         TimeSchedulerService timeSchedulerService) {
     this.authorizationEnforcer = authorizationEnforcer;
     this.authenticationContext = authenticationContext;
     this.scheduler = scheduler;
+    this.timeSchedulerService = timeSchedulerService;
+  }
+
+  /**
+   * Get the previous run time for the program. A program may contain one or more schedules
+   * the method returns the previous runtimes for all the schedules. This method only takes
+   + into account schedules based on time. For schedules based on data, an empty list will
+   + be returned.
+   *
+   * @param progrmaId program to fetch the previous runtime.
+   * @return list of Scheduled runtimes for the program. Empty list if there are no schedules
+   *         or if the program is not found
+   * @throws SchedulerException on unforeseen error.
```

Comment:
when does it throw other Exceptions?

Reply:
When authorization failed. It was copied from other method, and apparently the javadoc of this class is not very good.